### PR TITLE
Throw an error if args contain reserved python keywords

### LINKED ISSUE #135127: ban native ops from having python reserved kwargs as names
cc @ezyang @bhosmer @bdhirsh @kadeng

### PR DESCRIPTION
**Pull request by ishon19**
This PR adds a check for reserved python keywords in the `torchgen/gen.py/error_check_native_functions`  function.

Fixes #135127

### CODE REVIEW

File: torchgen/gen.py [hunk 1]
```diff
@@ -268,6 +269,12 @@ def error_check_native_functions(funcs: Sequence[NativeFunction]) -> None:
                 f"{f.structured_delegate}, but {f.structured_delegate} is not marked as structured. "
                 f"Consider adding 'structured=True' to the delegated operator"
             )
+        # Check for reserved Python keywords
+        for arg in f.func.arguments.flat_all:
+            if arg.name in keyword.kwlist:
```
[the remark this answers is not in the document]
Comment (ishon19) [reply torchgen/gen.py:274]: Sure, thanks for the review @Skylion007.

**Review by ezyang — APPROVED** [review]

**Comment by ezyang** [comment]
Not sure if this is passing CI though, I think we goofed a few schemas

**Comment by ishon19** [comment]
> Not sure if this is passing CI though, I think we goofed a few schemas

Thanks for the review @ezyang! Yeah, I can see a lot of checks failing 👀

**Comment by ezyang** [comment]
Probably the easiest thing is to grandfather in the preexisting violations

**Comment by ishon19** [comment]
> Probably the easiest thing is to grandfather in the preexisting violations

My bad, I was just trying to follow the build error stacktrace, I'll revert the last change. Need your suggestion on whether I should just add an exclusion for `from` in the `gen.py` file?

**Comment by ezyang** [comment]
I think just having a list literally at the spot you're doing the check is fine.

**Comment by Skylion007** [comment]
See https://docs.python.org/3/library/keyword.html keyword.kwlist and keyword.softkeyword list etc to initialize the list

**Comment by ezyang** [comment]
BTW, the exclusion should be specifically for the *operator and keyword*, not just the keyword, we shouldn't let other people add from keywords!

**Comment by ishon19** [comment]
Makes a lot of sense, thanks for the suggestions! I guess we can have a list of `(op_name, param_name)` tuples for the exclusion check.

**Review by ezyang — APPROVED** [review]

**Comment by ishon19** [comment]
@ezyang sorry to ping you again, but I fixed the lint errors flagged in the previous workflow run and I hope it's good to go now🤞

**Comment by ishon19** [comment]
Apologies for the repeated `lint` errors, ran the `lintrunner` this time around and hope it's fine now!

**Pull request closed without merging**